guardian/invoicing-api
# Next invoice date should not depend on the existance of current invoice period

## What does this change?

It attempts to fix a bug where if billing-preview does not return a *current* invoice period, that is, the first invoice period is actually represents the *next* one, then returning the one after would actually represent us skipping an invoice.

## How to test

I've added a test based on an issue identified in PROD by currently running [test in production ](https://github.com/guardian/support-service-lambdas/blob/7f7c3d340cfd99396e6efdef2f26924989cae3c4/lib/credit-processor/src/main/scala/com/gu/creditprocessor/Processor.scala#L117) which logs something like

```
ERROR - testInProdNextInvoiceDate failed because Add(*********,2020-11-09,2020-11-09,2020-11-09,List(ChargeOverride(*********,2020-10-24,2020-10-24,-3.68))) =/= 2020-12-09
```


We can continue to observe the results of this test.




## How can we measure success?

Less `testInProdNextInvoiceDate` logs.

## Have we considered potential risks?

No major risk because although test is running in production we have not fully switched over to the new behaviour.
